aten/src/ATen/detail/VULKANGuardImpl.cpp

```diff
@@ -0,0 +1,9 @@
+#include <ATen/detail/VULKANGuardImpl.h>
+
+namespace at {
+namespace detail {
+
+C10_REGISTER_GUARD_IMPL(VULKAN, VULKANGuardImpl);
```

Comment:
My capitalization here was based as in DeviceType.h all DeviceTypes are capitalized, so I added it as VULKAN and Device is VULKAN as , GUARD_IMPL is for Device and I used the same capitalization here. I prefer 'Vulkan' everywhere - in Device,DeviceType and here. Do you think it will be ok to have DeviceType 'Vulkan'?